.github/workflows/appimage.yml

```diff
@@ -72,15 +85,53 @@ jobs:
       run: |
         sudo -u testbot bash -lc 'cd rdk/services/slam/builtin && sudo go test -v -run TestCartographerIntegration'
 
-    - name: Build AppImage
-      run: sudo -u testbot bash -lc 'cd viam-cartographer && make BUILD_CHANNEL="latest" appimage'
-
     - name: Authorize GCP Upload
       uses: google-github-actions/auth@v1
       with:
         credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
 
-    - name: Deploy Files
+    - name: Build AppImage (PR)
+      if: contains(github.event.pull_request.labels.*.name, 'appimage') || contains(github.event.pull_request.labels.*.name, 'appimage-ignore-tests')
+      run: |
+         sudo -u testbot bash -lc 'cd viam-cartographer && make BUILD_CHANNEL="pr-${{ github.event.pull_request.number }}" appimage'
+
+    - name: Upload Files (PR)
+      if: contains(github.event.pull_request.labels.*.name, 'appimage') || contains(github.event.pull_request.labels.*.name, 'appimage-ignore-tests')
+      uses: google-github-actions/upload-cloud-storage@v0.10.4
+      with:
+        headers: "cache-control: no-cache"
+        path: 'viam-cartographer/etc/packaging/appimages/deploy/'
+        destination: 'packages.viam.com/apps/slam-servers/'
+        glob: '*'
+        parent: false
+        gzip: false
+
+    - name: Add AppImage Links (PR)
+      if: matrix.platform == 'linux/amd64' && (contains(github.event.pull_request.labels.*.name, 'appimage') || contains(github.event.pull_request.labels.*.name, 'appimage-ignore-tests'))
+      uses: marocchino/sticky-pull-request-comment@v2.2.0
+      with:
+        header: AppImage (x86_64)
+        number: ${{ env.PR_NUMBER }}
+        recreate: true
+        message: |
+           AppImage ready (x86_64): <https://packages.viam.com/apps/slam-servers/carto_grpc_server-pr-${{ github.event.pull_request.number }}-x86_64.AppImage>
+
+    - name: Add AppImage Links2 (PR)
+      if: matrix.platform == 'linux/arm64' && (contains(github.event.pull_request.labels.*.name, 'appimage') || contains(github.event.pull_request.labels.*.name, 'appimage-ignore-tests'))
+      uses: marocchino/sticky-pull-request-comment@v2.2.0
+      with:
+        header: AppImage (arm64)
+        number: ${{ env.PR_NUMBER }}
+        recreate: true
+        message: |
+          AppImage ready (arm64): <https://packages.viam.com/apps/slam-servers/carto_grpc_server-pr-${{ github.event.pull_request.number }}-aarch64.AppImage>
+
+    - name: Build AppImage (Latest)
+      if: github.event_name == 'workflow_dispatch' || github.event_name == 'push'
```

Comment:
Is this mutually exclusive with https://github.com/viamrobotics/viam-cartographer/pull/46/files#diff-00f0606e18b2618c3cc1667ca7c2b703b537af690ca71eba1330633587dcb1eeR94?

Reply:
Yes, these are mutually exclusive as 'workflow_dispatch' is a manual trigger from github.com, 'push' is the github event of a PR being merged and 'pull_request_target' is when a PR has been open and interacted with in one of several defined ways (labeled, opened, reopened, synchronized...). All events are mutually exclusive and if not, the concurrency group at the top of the pullrequest-trusted in concert with the !cancelled() check will prevent multiple runs from occurring simultaneously on the same PR.